### ISSUE TITLE
pl$DataFrame() fails when columns of type "ivs_iv" is present in dataframe

### ISSUE THREAD

**Issue by cathblatter** [pola-rs/r-polars#368]
Hi - came across an issue with daterange-columns from the [{ivs}](https://github.com/DavisVaughan/ivs) package in my use of {tidypolars}, as per the [author's suggestion](https://github.com/etiennebacher/tidypolars/issues/41#issuecomment-1695661523) I'm posting here.

In brief: `pl$DataFrame()` fails when an interval-column (daterange) using the {ivs}-package was added to the dataframe prior to converting, with the following error: 

```
library(dplyr, warn.conflicts = FALSE)
library(ivs)
library(polars)

t_date <- as.Date("2020-05-05")

test_df <- tibble(id = 1:5, 
                   grp = c("a", "a", "b", "b", "b"),
                   start = rep(t_date+1:5),
                   end = rep(t_date+11:7))

# adding an iv-variable to the dataframe
test_df_iv <- test_df |> 
    mutate(range = ivs::iv(start, end))

pl$DataFrame(test_df_iv)
#> Error: in set_column_from_robj: ShapeMismatch(ErrString("unable to add a column of length 2 to a dataframe of height 5")) 
```

I'm aware its a bit of a special column type and I am just curious if you plan to support these type of variables at some point? I find it extremely convenient to use r-polars with large routine data for the time saved (and it often contains date range data).

let me know if you want me to provide more info - in the meantime 🙌  thanks for the work!

**Comment by eitsupi** [comment]
It seems `ivs` is based on the `vctrs` package, and the `arrow` package already support `vctrs` package's class.

```r
> pak::pak("ivs")
✔ Updated metadata database: 2.90 MB in 6 files.                          
✔ Updating metadata database ... done                                     
                                                                           
→ Will install 1 package.
→ Will download 1 package with unknown size.
+ ivs   0.2.0 [dl]
ℹ Getting 1 pkg with unknown size
✔ Got ivs 0.2.0 (x86_64-pc-linux-gnu-ubuntu-22.04) (412.73 kB)     
✔ Downloaded 1 package (412.73 kB)in 3.5s                          
✔ Installed ivs 0.2.0  (62ms)                               
✔ 1 pkg + 5 deps: kept 5, added 1, dld 1 (412.73 kB) [18.4s]                                  

> library(dplyr)

Attaching package: ‘dplyr’

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union


> library(ivs)

> t_date <- as.Date("2020-05-05")
 
    test_df <- tibble(id = 1:5, 
                       grp = c("a", "a", "b", "b", "b"),
                       start = rep(t_date+1:5),
                       end = rep(t_date+11:7))
 
    # adding an iv-variable to the dataframe
    test_df_iv <- test_df |> 
        mutate(range = ivs::iv(start, end))

> test_df_iv$range
<iv<date>[5]>
[1] [2020-05-06, 2020-05-16) [2020-05-07, 2020-05-15) [2020-05-08, 2020-05-14) [2020-05-09, 2020-05-13)
[5] [2020-05-10, 2020-05-12)

> test_df_iv$range |> class()
[1] "ivs_iv"     "vctrs_rcrd" "vctrs_vctr"

> test_df_iv |> arrow::as_arrow_table()
Table
5 rows x 5 columns
$id <int32>
$grp <string>
$start <date32[day]>
$end <date32[day]>
$range <<iv<date>[0]>>

> test_df_iv |> arrow::as_arrow_table() |> _$range
ChunkedArray
<<iv<date>[0]>>
[
  -- is_valid: all not null
  -- child 0 type: date32[day]
    [
      2020-05-06,
      2020-05-07,
      2020-05-08,
      2020-05-09,
      2020-05-10
    ]
  -- child 1 type: date32[day]
    [
      2020-05-16,
      2020-05-15,
      2020-05-14,
      2020-05-13,
      2020-05-12
    ]
]
```

But when I try to convert this to polars I get an error. Perhaps the `arrow2` crate does not support this type.

In other words, it's likely an upstream issue.

```r
> test_df_iv |> arrow::as_arrow_table() |> polars::pl$from_arrow()
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Utf8Error', /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow2-0.17.4/src/ffi/schema.rs:501:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'explicit panic', src/rdataframe/mod.rs:82:1
Error: Execution halted with the following contexts
   0: In R: in pl$from_arrow:
   0: During function call [polars::pl$from_arrow(arrow::as_arrow_table(test_df_iv))]
   1: user function panicked: from_arrow_record_batches
```

**Comment by sorhawell** [comment]
I have fix for this upcomming, but got interrupted. Will update later.

**Comment by sorhawell** [comment]
Hi @cathblatter from released polars 0.8.0 with PR #369 above example has been enabled. Should we close this issue?

**Comment by cathblatter** [comment]
Confirm this works like a charm -  thank you very much @sorhawell  & also @etiennebacher for your speedy adjustments🥳
